### PR TITLE
Fixed all (most) weapons wielded sprite handling.

### PR DESCRIPTION
**Pull request by Deal5**
## About The Pull Request

Fixed all listed weapons twohanded state handling. (IF NOT ALL)

All listed weapons no longer dissapear after being taken in two hands, laser weapons that have charge indicator get updated as they fire, some sprites are stolen from other guns, some are stolen and redrawn (lightfall, harpoon gun (Idc its not a weapon) as example), some weapons that had _do(u)ble sprites now have them (vintorez)

The only thing I can't fix right now is how sprite isnt updated when gun mods that change visual appearance don't update the item immediately (silencers, gold paint mostly)

## Why It's Good For The Game

People will not shoot with their hands anymore (perhaps).
Everything should work as intended (surely).

## Testing

Local server, tested every _listed_ weapon personally (pain). 

## Changelog
:cl:
add: harpoon gun sprite
fix: most ranged weapons should work as intended when being taken in two hands
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/stranger.dm [hunk 1]
```diff
@@ -137,27 +137,28 @@
 		list(mode_name="bP6hfnj3Js", mode_desc="AhG8GjobYa", burst=3, projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=5, move_delay=4, charge_cost=11, icon="vaporize", projectile_color = "#FFFFFF")
 	)
 
-/obj/item/gun/energy/plasma/stranger/update_icon(ignore_inhands)
+/obj/item/gun/energy/plasma/stranger/update_icon(var/ignore_inhands)
```
Comment (TheShown911) [inline code/modules/mob/living/simple_animal/hostile/stranger.dm:140]: I think we stopped doing var/ in arguments.

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -213,37 +212,38 @@
...
+            else
+                item_state_slots[slot_l_hand_str] = "lefthand"
+                item_state_slots[slot_r_hand_str] = "righthand"
+        else//Otherwise we can just pull from the generic left and right hand icons.
+            if(wielded_icon)
+                item_state_slots[slot_l_hand_str] = wielded_item_state
+                item_state_slots[slot_r_hand_str] = wielded_item_state
+            else
+                item_state_slots[slot_l_hand_str] = initial(item_state)
+                item_state_slots[slot_r_hand_str] = initial(item_state)
```
Comment (TheShown911) [inline code/modules/projectiles/gun.dm:246]: Wh-why would you replace tabs with spaces?

File: code/modules/projectiles/guns/energy/plasma.dm [hunk 3]
```diff
@@ -85,9 +84,22 @@
 	init_recoil = RIFLE_RECOIL(1)
 	spawn_tags = SPAWN_TAG_FS_ENERGY
 
-/obj/item/gun/energy/plasma/cassad/update_icon()
+/obj/item/gun/energy/plasma/cassad/update_icon(var/ignore_inhands)
```
Comment (TheShown911) [inline code/modules/projectiles/guns/energy/plasma.dm:87]: Here too.

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 4]
```diff
@@ -92,3 +93,18 @@
...
+		//make sure that rounding down will not give us the empty state even if we have charge for a shot left.
+		if(cell && cell.charge >= charge_cost)
+			ratio = cell.charge / cell.maxcharge
+			ratio = min(max(round(ratio, 0.25) * 100, 25), 100)
+		if(item_charge_meter)
+			set_item_state("-[ratio]")
+			wielded_item_state = "_doble" + "-[ratio]"
+			icon_state = initial(icon_state) + "[ratio]"
+	if(!ignore_inhands)
+		update_wear_icon()
```
Comment (TheShown911) [inline code/modules/projectiles/guns/energy/temperature.dm:110]: No newline

**Review by TheShown911 — APPROVED** [review]
Extremely based PR.

**Review by hyperioo — CHANGES_REQUESTED** [review]
See Shown's comments + Minor conflict (can be solved with web editor, it's just one line)

**Review by hyperioo — APPROVED** [review]
👍

**Pull request merged**